managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -352,6 +352,7 @@ public void asyncOpen(final String name, final ManagedLedgerConfig config, final
                         // Managed ledger is in unusable state. Recreate it.
                         log.warn("[{}] Attempted to open ledger in {} state. Removing from the map to recreate it",
                                 name, l.getState());
+                        l.close();
```

Comment:
Should it be called asynchronously after the removal from the map? What if close blocks or throws an exception?

Reply:
@lhotari please take a look again